src/common/utils/url.ts

```
@@ -184,6 +184,28 @@ function cleanPathName(basePathName: string, pathName: string) {
     return pathName;
 }
 
+function isCallsPopOutURL(serverURL: URL | string, inputURL: URL | string, callID: string) {
+    if (!serverURL || !inputURL || !callID) {
+        return false;
+    }
+
+    const parsedURL = parseURL(inputURL);
```

`isURLType` already does this check. I think we can try and reduce the amount of checks we're doing here to avoid some bloat.

---

@devinbinnie  `isTeamUrl` does the same really. We need to parse to strip the server portion. We could modify `isUrlType` to avoid parsing the input if already an URL and potentially receive the serverInfo object directly. It feels a bit out of scope but happy to do it in here. Also happy to hear alternatives of course.

---

I was thinking we don't have to do this check twice:
https://github.com/mattermost/desktop/blob/7cfeebf2a27abd5e1b068364d616e0cb75f187a2/src/common/utils/url.ts#L89
https://github.com/mattermost/desktop/blob/7cfeebf2a27abd5e1b068364d616e0cb75f187a2/src/common/utils/url.ts#L194

---

I guess I was trying to fail early if the server is not matching, same as in `isTeamUrl` but yeah, we could also defer it. Not sure it will improve the bloating that match but it's one less duplicated check in the happy path. I'll have another look.

---

No worries, should be done. I simplified `isTeamUrl` as well since I was there.